modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -293,6 +295,34 @@ public ObjectRelationship deleteObjectRelationship(
 				"Reverse object relationships cannot be deleted");
 		}
 
+		for (long objectFieldId :
+				(List<Long>)objectRelationshipPersistence.dslQuery(
+					DSLQueryFactoryUtil.selectDistinct(
+						ObjectFieldSettingTable.INSTANCE.objectFieldId
+					).from(
+						ObjectRelationshipTable.INSTANCE
+					).innerJoinON(
+						ObjectFieldTable.INSTANCE,
+						ObjectFieldTable.INSTANCE.objectDefinitionId.eq(
```

Comment:
I believe we need to add a condition for `objectDefinitionId2` to cover the Many to Many case. Could you please check this?

Feel free to ask if you have any question